src/network/network_osm_reader.cpp

```diff
@@ -0,0 +1,361 @@
+// network_osm_reader.cpp -- function to read OSM data as network
+
+#include "network/network.hpp"
+#include "util/debug.hpp"
+
+#include <unordered_map>
+#include <unordered_set>
+
+#include <osmium/handler.hpp>
+#include <osmium/io/any_input.hpp>
+#include <osmium/osm/node.hpp>
+#include <osmium/osm/way.hpp>
+#include <osmium/visitor.hpp>
+
+
+using namespace FMM;
+using namespace FMM::CORE;
+using namespace FMM::MM;
+using namespace FMM::NETWORK;
+
+
+
+/** Helper class to filter out OSM ways that are not roads */
+class way_filter {
+protected:
+  /** key -- value pairs; if any of the keys is present with any
+   * of the corresponding values, this item is skipped */
+  std::unordered_map<std::string, std::unordered_set<std::string> > exclude_tags;
+  /** key -- value pairs; all of the keys here need to be present
+   * with any of the values given -- if no value given, the key
+   * has to be present with any value */
+  std::unordered_map<std::string, std::unordered_set<std::string> > include_tags;
+
+public:
+  /** test if the given way matches the filters */
+  bool operator () (const osmium::Way& way) const {
+		/* test include_tags first -- note that this is an O(n*m) operation
+     * where n is the size of include_tags, and m is the number of tags
+     * the way has; since the size of include_tags will be typically
+     * small (1), this will not be a practical problem */
+		for(const auto& p : include_tags) {
```

Comment:
As before: if a way does not contain an include tag, it should be filtered out (the reasoning is the all roads should have the "highway" tag).
Either we want to filter for specific values or just the general presence of a tag with any value. I agree this is too general for this case, so it can be a good idea to simplify it.